### PR TITLE
Hotfix/fix url to path func

### PR DESCRIPTION
**Pull request by rossmiller**
This fix removes some redundant util logic - we don't need the remove signature logic anymore. It also fixes an issue where non private acl bucket content links were not rendered. A test has been added for different types of S3 URL to make sure the util func can handle them.

### CODE REVIEW

File: s3upload/utils.py [hunk 1]
```diff
@@ -201,26 +191,3 @@ def get_signed_download_url(
     )
 
     return download_url
-
-
```
Comment (rossmiller) [inline s3upload/utils.py]: Removing remove_signature - it's not needed in the util func that was using it (it was never needed)

File: s3upload/widgets.py [hunk 2]
```diff
@@ -43,10 +43,14 @@ def get_file_url(self, value):
...
-            return get_signed_download_url(
-                value,
-                bucket_name,
-            ) if self.acl == 'private' else value
+            if self.acl == 'private':
+                return get_signed_download_url(
+                    value,
+                    bucket_name,
+                )
+            else:
```
Comment (rossmiller) [inline s3upload/widgets.py]: We need to return the URL for the resource here or non private acl configurations provide the incorrect resource address.

File: s3upload/utils.py [hunk 3]
```diff
@@ -166,19 +164,11 @@ def create_upload_data(content_type, key, acl, bucket=None, cache_control=None,
...
-    path = path.split(bucket_name)[-1]
-    return path
-
-
-def get_key_from_url(url, bucket_name=settings.AWS_STORAGE_BUCKET_NAME):
-    conn = boto.connect_s3()
-    bucket = conn.get_bucket(bucket_name)
-    path = get_path_from_url(url)
-    return bucket.get_key(path)
+    return path.split(bucket_name)[-1]
```
Comment (rossmiller) [inline s3upload/utils.py]: I spent a while reviewing this logic and realised that there was a lot of unnecessary hoop jumping. I've simplified it and verified it manually and with the new unit tests.

File: s3upload/tests.py [hunk 4]
```diff
@@ -176,17 +174,33 @@ def test_content_length_range(self):
 
 
 class UtilsTest(TestCase):
-    def test_remove_signature(self):
-        test_url = "https://test-bucket.s3.amazonaws.com/image.jpg"
+    def test_get_s3_path_from_url(self):
```
Comment (rossmiller) [inline s3upload/tests.py]: New unit test for util function explained below.

File: s3upload/utils.py [hunk 5]
```diff
@@ -166,19 +164,11 @@ def create_upload_data(content_type, key, acl, bucket=None, cache_control=None,
...
-def get_path_from_url(url, bucket_name=settings.AWS_STORAGE_BUCKET_NAME):
+def get_s3_path_from_url(url, bucket_name=settings.AWS_STORAGE_BUCKET_NAME):
     path = urlparse(url).path
     # The bucket name might be part of the path,
     # so get the path that comes after the bucket name
-    path = path.split(bucket_name)[-1]
-    return path
-
-
-def get_key_from_url(url, bucket_name=settings.AWS_STORAGE_BUCKET_NAME):
```
Comment (rossmiller) [inline s3upload/utils.py]: This is totally redundant now.

File: s3upload/fields.py [hunk 6]
```diff
@@ -29,12 +29,13 @@ def pre_save(self, model_instance, add):
 
         if file_url:
             setattr(model_instance, self.attname, file_url)
-            no_signature_url = remove_signature(file_url)
             bucket_name = settings.S3UPLOAD_DESTINATIONS[self.dest].get(
                 'bucket', settings.AWS_STORAGE_BUCKET_NAME
             )
-            path = get_path_from_url(no_signature_url, bucket_name=bucket_name)
-            return path
+            return get_s3_path_from_url(
```
Comment (rossmiller) [inline s3upload/fields.py]: Call to renamed util func (see below). This uses "urlparse" which separates out query strings anyway, so we can provide a URL with a query string - no need for "remove_signature".

File: s3upload/tests.py [hunk 7]
```diff
@@ -176,17 +174,33 @@ def test_content_length_range(self):
...
+        )
+        test_s3_url_3 = 'https://{0}.s3.amazonaws.com:443/folder1/folder2/file1.json'.format(
+            settings.AWS_STORAGE_BUCKET_NAME
+        )
+        test_s3_url_4 = 'https://s3-aws-region.amazonaws.com/{0}/folder1/folder2/file1.json'.format(
+            settings.AWS_STORAGE_BUCKET_NAME
+        )
+        test_s3_url_5 = 'https://s3-aws-region.amazonaws.com:443/{0}/folder1/folder2/file1.json'.format(
+            settings.AWS_STORAGE_BUCKET_NAME
+        )
```
Comment (stevejalim) [inline s3upload/tests.py]: Nice coverage 👍

**Pull request merged**